src/reference/01-General-Info/90-Changes/45-Migrating-from-sbt-0.12.x.md

```diff
@@ -130,3 +139,84 @@ when migrating you mustn't use `.value` but `.evaluated`:
 ```scala
 run := docsRunSetting.evaluated
 ```
+
+### Migrating from the Build trait
+
+With `Build` trait based build such as:
+
+```scala
+import sbt._
+import Keys._
+import xyz.XyzPlugin.autoImport._
+
+object HelloBuild extends Build {
+  val shared = Defaults.defaultSettings ++ xyz.XyzPlugin.projectSettings ++ Seq(
+    organization := "com.example",
+    version      := "0.1.0",
+    scalaVersion := "2.12.1")
+
+  lazy val hello =
+    Project("Hello", file("."),
+      settings = shared ++ Seq(
+        xyzSkipWrite := true)
+    ).aggregate(core)
+
+  lazy val core =
+    Project("hello-core", file("core"),
+      settings = shared ++ Seq(
+        description := "Core interfaces",
+        libraryDependencies ++= scalaXml.value)
+    )
+
+  def scalaXml = Def.setting {
+    scalaBinaryVersion.value match {
+      case "2.10" => Nil
+      case _      => ("org.scala-lang.modules" %% "scala-xml" % "1.0.6") :: Nil
+    }
+  }
+}
+```
+
+You can migrate to `build.sbt`:
+
+```scala
+val shared = Seq(
+  organization := "com.example",
+  version      := "0.1.0",
+  scalaVersion := "2.12.1"
+)
+
+lazy val helloRoot = (project in file("."))
+  .aggregate(core)
+  .enablePlugins(XyzPlugin)
+  .settings(
+    shared,
+    name := "Hello",
+    xyzSkipWrite := true
+  )
+
+lazy val core = (project in file("core"))
+  .enablePlugins(XyzPlugin)
+  .settings(
+    shared,
+    name := "hello-core",
+    description := "Core interfaces",
+    libraryDependencies ++= scalaXml.value
+  )
+
+def scalaXml = Def.setting {
+  scalaBinaryVersion.value match {
+    case "2.10" => Nil
+    case _      => ("org.scala-lang.modules" %% "scala-xml" % "1.0.6") :: Nil
+  }
+}
+```
+
+1. Rename `project/Build.scala` to `build.sbt`.
+2. Remove import statements `import sbt._`, `import Keys._`, and any auto imports.
+3. Move all of the inner definitions (like `shared`, `helloRoot`, etc) out of the `object HelloBuild`, and remove `HelloBuild`.
+4. Change `Project(...)` to `(project in file("x"))` style, and call its `settings(...)` method to pass in the settings. This is so the auto plugins can reorder their setting sequence based on the plugin dependencies. `name` setting should be set to keep the old names.
```

Comment:
for sbt 1 `settings(...)` became lazy in https://github.com/sbt/sbt/pull/2543 to fix https://github.com/sbt/sbt/issues/2104 reported by Scala.js. I think there is a workaround to define `LocalProject("appJVM")`. We can smooth the migration ramp a bit by potentially providing `lazySettings(...)` method in sbt 0.13.x series.